packages/next/src/server/lib/server-action-request-meta.ts

```diff
@@ -34,14 +35,21 @@ export function getServerActionRequestMetadata(
       req.method === 'POST'
   )
 
-  return { actionId, isURLEncodedAction, isMultipartAction, isFetchAction }
+  const isServerAction = Boolean(
+    isFetchAction || isURLEncodedAction || isMultipartAction
+  )
+
+  return {
+    actionId,
+    isURLEncodedAction,
+    isMultipartAction,
+    isFetchAction,
+    isServerAction,
+  }
 }
 
 export function getIsServerAction(
   req: IncomingMessage | BaseNextRequest | NextRequest
 ): boolean {
-  const { isFetchAction, isURLEncodedAction, isMultipartAction } =
-    getServerActionRequestMetadata(req)
-
-  return Boolean(isFetchAction || isURLEncodedAction || isMultipartAction)
+  return getServerActionRequestMetadata(req).isServerAction
 }
```

Comment:
This change avoids calling `getServerActionRequestMetadata` twice in the Action's handler.